sympy/solvers/solvers.py

```diff
@@ -2301,162 +2301,22 @@ def solve_linear_system(system, *symbols, **flags):
     {}
 
     """
-    do_simplify = flags.get('simplify', True)
+    from sympy.solvers.solveset import linsolve
+    from sympy.sets import FiniteSet
 
-    if system.rows == system.cols - 1 == len(symbols):
```

Comment:
Perhaps this case should be moved to linsolve; there are optimizations in `inv_quick` that should probably be kept.

Reply:
Maybe. I think that we can make rref much faster though by using the poly domains (https://github.com/sympy/sympy/issues/18445#issuecomment-578286594)

Reply:
Until then, if my memory serves correctly, the `inv_quick` was significantly faster. I thought I added it after there was an issue raised abotu how slow solving relatively small symbolic systems was.
```python
>>> t=time();ok=inv_quick(Matrix(((x/2,y),(z,2*x))));time()-t
0.00700068473815918
>>> t=time();ok=Matrix(((x/2,y),(z,2*x))).inv();time()-t
0.41904187202453613
```